File: backend/src/main/java/com/g0urd/grad_project/service/CargoService/CargoService.java
package com.g0urd.grad_project.service.CargoService;

import java.util.List;

import com.g0urd.grad_project.models.Cargo;

public interface CargoService {

    List<Cargo> fetchAllCargos();

    Cargo findById(Long id);

    Cargo createCargo(Cargo cargo);

    Cargo updateCargo(Cargo cargo);

    Boolean deleteCargo(Long id);

}

File: frontend/src/app/service/cargo/cargo.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Cargo } from '../../component/cargo/cargo';
import { Observable } from 'rxjs';

const BASE_URL = ["http://localhost:8080/api/manager"]

@Injectable({
  providedIn: 'root'
})
export class CargoService {

  constructor(private http: HttpClient) { }

  fetchAllCargos():Observable<Cargo[]> {
    return this.http.get<Cargo[]>(BASE_URL + `/cargo`);
  }

  createCargo(data: Cargo) {
    return this.http.post<Cargo>(BASE_URL + `/cargo`, data);
  }

  updateCargo(data: Cargo) {
    return this.http.put<Cargo>(BASE_URL + `/cargo/${data.id_cargo}`, data);
  }

  deleteCargo(id_cargo: Number) {
    return this.http.delete<Cargo>(BASE_URL + `/cargo/${id_cargo}`);
  }
}
